### DISCUSSION TITLE
Hitting Breakpoints in Modules

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#757] [Q&A]
We are adding some new folks to our project and I'm looking for the most straightforward way to debug modules.  I'm currently using the add-module-projects-to-octane-solution but that has always felt a bit hacky.  Is there another way or is this the simplest approach?

**Comment by hishamco** [comment]
What if you debug it directly from the Oqtane with source linking enabled?

**Reply by markdav-is** [reply]
not sure how to det a breakpoint that way, but it should allow me to step-in to code.

**Reply by markdav-is** [reply]
thanks for that tip!  it's hitting my breakpoints!

**Comment by sbwalker** [comment]
The recommended approach to building modules is to use the external module approach so that your code is physically separated from the core framework. The best article on debugging external module is from @ADefWebserver https://blazorhelpwebsite.com/ViewBlogPost/4 - scroll down to the section for "Debugging External Oqtane Modules"

**Reply by hishamco** [reply]
I tried the mentioned way but the break point doesn't hit with Oqtane Blog module, did I miss something else

/cc @ADefWebserver

**Comment by ADefWebserver** [comment]
It worked for me :)

**Reply by hishamco** [reply]
I don't why the break point doesn't hit :(
